src/dp_util.c

```diff
@@ -160,3 +166,45 @@ void dp_free_jhash_table(struct rte_hash *table)
 {
 	rte_hash_free(table);
 }
+
+int dp_set_vf_rate_limit(uint16_t port_id, uint64_t rate)
+{
+	char filename[DP_SYSFS_MAX_PATH];
+	uint16_t vf_pattern_len = 0;
+	const char *pattern = dp_conf_get_vf_pattern();
+	FILE *fp;
+	struct dp_port *port = dp_get_port_by_id(port_id);
+	uint64_t rate_in_mbits = rate;
+
+	if (!port) {
+		DPS_LOG_ERR("Cannot get port by id", DP_LOG_PORTID(port_id));
+		return DP_ERROR;
+	}
+
+	while (*(pattern + vf_pattern_len) != '\0')
+		vf_pattern_len++;
+
+	if (snprintf(filename, sizeof(filename),
+				"%s%s%s%s%s",
+				DP_SYSFS_PREFIX_MLX_DEVICE,
+				dp_conf_get_pf0_name(),
+				DP_SYSFS_PREFIX_MLX_MAX_TX_RATE,
+				port->vf_name + vf_pattern_len,
+				DP_SYSFS_SUFFIX_MLX_MAX_TX_RATE)
+			>= (int)sizeof(filename)
+	) {
+		DPS_LOG_ERR("SR-IOV sysfs path to vf's max tx rate is too long");
+		return DP_ERROR;
+	}
```

Comment:
The better check is to actually cast `(unsigned)printf()` because there can be a negative result (encoding error) and by casting it to unsigned, you force it to be higher than `sizeof(filename)`

Reply:
used a more explicit way to check and also adapt another location with the same issue.